### PR TITLE
SSX React: Add additional provider options

### PR DESCRIPTION
**Pull request by skgbafa**
# Description

This updates the ssx-react API to allow for additional options to pass to the ssx instance. It also updates the react examples. There are now two react examples using [wagmi](https://wagmi.sh) and [Blocknative's web3-onboard](https://onboard.blocknative.com/) to get the provider and pass it to SSX

# Type

- [x] New feature (non-breaking change which adds functionality)

# Diligence Checklist
- [x] This change requires a documentation update
- [x] I have updated and/or included new end-to-end tests
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/ssx-react/src/ssx.tsx [hunk 1]
```diff
@@ -2,12 +2,22 @@ import { useContext, createContext, useState, useEffect, ReactNode } from 'react
...
+}
+
 /** Props for SSX Provider. */
 export interface SSXProviderProps {
   /** Optional SSX configuration, used for instantiating an SSX Instance. */
   ssxConfig?: SSXClientConfig;
   /** Provider child nodes, for rendering. */
   children: ReactNode;
+  /** Optional SSX Signer. Will . */
+  web3Provider?: SSXWeb3Provider;
```
Comment (samgbafa) [inline packages/ssx-react/src/ssx.tsx]: @w4ll3 @Juliano1612 Should I move these to ssx-core? They are really only used in ssx-react

Comment (w4ll3) [reply packages/ssx-react/src/ssx.tsx]: I suppose it's fine to keep it then.

Comment (Juliano1612) [reply packages/ssx-react/src/ssx.tsx]: Agree with @w4ll3

File: examples/ssx-test-onboard-react-dapp/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "with-nextjs",
```
Comment (Juliano1612) [inline examples/ssx-test-onboard-react-dapp/package.json]:
```suggestion
  "name": "ssx-test-onboard-react-dapp",
```

File: examples/ssx-test-onboard-react-dapp/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
```
Comment (Juliano1612) [inline examples/ssx-test-onboard-react-dapp/README.md]: We can add here more context about the example and the instructions to populate the .env and run the application

File: examples/ssx-test-wagmi-react-dapp/package.json [hunk 4]
```diff
@@ -25,4 +25,4 @@
     "eslint-config-next": "^12.1.6",
     "typescript": "^4.7.2"
   }
-}
\ No newline at end of file
+}
```
Comment (Juliano1612) [inline examples/ssx-test-wagmi-react-dapp/package.json:28]: Update the project name to `ssx-test-wagmi-react-dapp`

**Review by Juliano1612 — CHANGES_REQUESTED** [review]

File: examples/ssx-test-wagmi-react-dapp/package.json (hunk 4 above)
Comment (samgbafa) [reply examples/ssx-test-wagmi-react-dapp/package.json:28]: Done

File: examples/ssx-test-onboard-react-dapp/README.md (hunk 3 above)
Comment (samgbafa) [reply examples/ssx-test-onboard-react-dapp/README.md]: Done

File: examples/ssx-test-onboard-react-dapp/package.json (hunk 2 above)
Comment (samgbafa) [reply examples/ssx-test-onboard-react-dapp/package.json]: Done

**Review by w4ll3 — APPROVED** [review]
lgtm

**Review by Juliano1612 — APPROVED** [review]
Looks good to me!

**Pull request merged**